alacritty/src/migrate/yaml.rs

```diff
@@ -0,0 +1,87 @@
+//! Migration of legacy YAML files to TOML.
+
+use std::path::Path;
+
+use toml::Value;
+
+use crate::cli::MigrateOptions;
+use crate::config;
+use crate::migrate::{migrate_config, migrate_toml, write_results};
+
+/// Migrate a legacy YAML config to TOML.
+pub fn migrate(
+    options: &MigrateOptions,
+    path: &Path,
+    recursion_limit: usize,
+    prefix: &str,
+) -> Result<String, String> {
+    // Try to parse the configuration file.
+    let mut config = match config::deserialize_config(path, !options.dry_run) {
+        Ok(config) => config,
+        Err(err) => return Err(format!("YAML parsing error: {err}")),
+    };
+
+    // Migrate config imports.
+    if !options.skip_imports {
+        migrate_imports(options, &mut config, path, recursion_limit)?;
+    }
```

Comment:
In theory it would probably be possible to simplify this even more, by removing all the YAML import code and just moving the YAML file immediately and then performing a TOML migration, however that will likely change the way output is done and it makes dry-runs and removal of those files in error scenarios more difficult.

So considering that YAML is on its way out anyway and we'll remove the migration for it in the future, I don't think it's worth it to risk breaking anything here.